server/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -74,7 +75,7 @@ public class RoutingNodes extends AbstractCollection<RoutingNode> {
 
     private int relocatingShards = 0;
 
-    private final Map<String, Set<String>> attributeValuesByAttribute;
+    private final ConcurrentMap<String, Set<String>> attributeValuesByAttribute;
```

Comment:
This can remain as a `Map` right?

```suggestion
    private final Map<String, Set<String>> attributeValuesByAttribute;
```